### ISSUE TITLE
Performing a prediction on a new user

### ISSUE THREAD

**Issue by SebastianLeborg** [benfred/implicit#152]
Hello!

I'm trying to assess the performance of my model. I split up my data into a test and train set I then plan to apply a mask which hides 20% of the test user's preferences in an iterative KFold fashion to measure the mean RMS (root mean square error) of the predictions on the test set.

I'm however struggling with performing predictions on new users. Going through the code, I figured that we can obtain the reconstruction of the training sets preferences by taking the dot product between the users and items latent factors. A new user, even with a history, of course doesn't have any latent factors. Am I missing something here? After pondering some more about this I figured that the latent factors are part of the reconstruction of the whole item-user preference matrix and is calculated during the training. Is it possible to obtain the latent factors of a new user without retraining the entire model? Or is this simply an impossibility when using ALS/collaborative filtering?

**Comment by benfred** [comment]
So this is possible! The 'recommend' call takes a sparse matrix of user_items. If you pass in the 'recalculate_user=True' to the recommend call it will calculate factors for the user on the fly - meaning that you can predict for new users (or for users that have updated preferences since originally training). Take a look at the unittest for an example: https://github.com/benfred/implicit/blob/master/tests/recommender_base_test.py#L47

For assessing the performance of the model, there is code in implicit.evaluation that might help you out (code for doing a train/test split and calculating some basic metrics).
